lib/manager/gradle/gradle-updates-report.ts

```diff
@@ -46,11 +48,19 @@ allprojects {
            .findAll { !it.startsWith('file:') }
            .unique()
         project.repositories = repos
-        def deps = (buildscript.configurations + configurations)
-          .collect { it.dependencies + it.dependencyConstraints }
-          .flatten()
-          .findAll { it instanceof DefaultExternalModuleDependency || it instanceof DependencyConstraint }
-          .collect { ['name':it.name, 'group':it.group, 'version':it.version] }
+        def deps = []
+        (buildscript.configurations + configurations)
+           .collect {
+               def name = it.name == 'classpath' ? 'plugins' : it.name
+               new Tuple2<String, List<Object>>(name, it.dependencies + it.dependencyConstraints)
+           }.groupBy { it.first }
+           .each { configuration, dependencies ->
+               def depsByConfig = dependencies
+                       .flatten()
+                       .findAll { it instanceof DefaultExternalModuleDependency || it instanceof DependencyConstraint }
+                       .collect { ['name': it.name, 'group': it.group, 'version': it.version, 'configuration': configuration] }
+               deps.push(depsByConfig)
```

Comment:
I think this is an easier way to do all of this:

```groovy
project.dependencies =  (buildscript.configurations + configurations)
    .collectMany { configuration -> 
        (configuration.dependencies + configuration.dependencyConstraints)
            .findAll { it instanceof DefaultExternalModuleDependency || it instanceof DependencyConstraint }
            .collect { ['name': it.name, 'group': it.group, 'version': it.version, 'configuration': configuration.name] }
    }
```

Reply:
Thank you, I really needed some Groovy help 😅